tests/test_parser.py

```diff
@@ -853,3 +853,24 @@ def test_parse_location():
     ctx = MLContext()
     attr = Parser(ctx, "loc(unknown)").parse_optional_location()
     assert attr == LocationAttr()
+
+
+@pytest.mark.parametrize(
+    "keyword,expected",
+    [
+        ("public", StringAttr("public")),
+        ("nested", StringAttr("nested")),
+        ("private", StringAttr("private")),
+        ("privateeee", None),
+        ("unknown", None),
+    ],
+)
+def test_parse_visibility(keyword: str, expected: StringAttr | None):
+    assert Parser(MLContext(), keyword).parse_optional_visibility_keyword() == expected
+
+    parser = Parser(MLContext(), keyword)
+    if expected is None:
+        with pytest.raises(ParseError):
+            parser.parse_visibility_keyword()
```

Comment:
Can you add a `match=` to the `pytest.raises()`?
It might sound overkill but I find it neat to catch error-reporting regressions, which otherwise are just noticed months later by confused users and are tricky to trace back.